core/src/main/java/data/movies/MoviesRepository.kt

```diff
@@ -60,4 +64,45 @@ class MoviesRepository @Inject constructor(
         }
 
     fun observeUpcomingMovies() = local.upcomingStore.observeEntries()
-}
\ No newline at end of file
+
+    // ------------- Recommendations capabilities -------------------
+    suspend fun saveRecommendations(page: Int, movies: List<Movie>) {
+        local.recommendationsStore.insert(page, movies)
+    }
+
+    suspend fun getRecommendations(movieId: Int) =
+        flow {
+            emit(remote.getRecommendations(movieId))
+        }
+
+    fun observeRecommendations() = local.recommendationsStore.observeEntries()
+
+    // ------------- Credits capabilities -------------------
+    suspend fun saveCredits(movieId: Int, actors: List<Actor>) {
+        local.creditsStore.insert(movieId, actors)
+    }
+
+    suspend fun getCredits(movieId: Int) =
+        flow {
+            emit(remote.getCredits(movieId))
+        }
+
+    fun observeCredits() = local.creditsStore.observeEntries()
+
+    // ------------- Movie Catch capabilities -------------------
+    fun getMovieFromCatch(movieId: Int) =
+        combine(
+            local.nowPlayingStore.observeEntries(),
+            local.popularStore.observeEntries(),
+            local.upcomingStore.observeEntries(),
+            local.topRatedStore.observeEntries()
+        ) { nowPlaying, popular, upcoming, topRated ->
+            mutableListOf<Movie>()
+                .addFlatLists(
```

Comment:
addFlatLists is redundant
you can use flatten operator

Reply:
ok